core/src/gl/framebuffer.cpp

```diff
@@ -72,6 +72,21 @@ GLuint FrameBuffer::readAt(float _normalizedX, float _normalizedY) const {
     return pixel;
 }
 
+FrameBuffer::PixelRect FrameBuffer::readRect(float _normalizedX, float _normalizedY, float _normalizedW, float _normalizedH) const {
+
+    PixelRect rect;
+    rect.left = floorf(_normalizedX * m_width);
+    rect.bottom = floorf(_normalizedY * m_height);
+    rect.width = ceilf(_normalizedW * m_width);
+    rect.height = ceilf(_normalizedH * m_height);
+
+    rect.pixels.resize(rect.width * rect.height);
+
+    GL::readPixels(rect.left, rect.bottom, rect.width, rect.height, GL_RGBA, GL_UNSIGNED_BYTE, rect.pixels.data());
```

Comment:
Checking the gl docs in a little bit more depth:
>with lower left corner at (x+i,y+j) for 0<=i<width and 0<=j<height.

`width` and `height` are under a lower strict; we should add `-1` to the `minf` to each of the bottom/left clamping equation since it seems that those conditions would still not be satisfied. Can you confirm that?